pkg/authorization/rewrite/rewrite.go

```diff
@@ -43,39 +44,58 @@ var _ authorizer.Authorizer = &rewritingAuthorizer{}
 func (n *rewritingAuthorizer) Authorize(ctx context.Context, attrs authorizer.Attributes) (authorizer.Decision, string, error) {
 	proxyAttrs := n.attributesGenerator.Generate(ctx, attrs)
 
+	// This should never happen, but we want to be sure.
 	if len(proxyAttrs) == 0 {
 		return authorizer.DecisionDeny,
 			"The request or configuration is malformed.",
 			fmt.Errorf("bad request. The request or configuration is malformed")
 	}
 
-	var (
-		authorized authorizer.Decision
-		reason     string
-		err        error
-	)
-
-	// AND logic on all SubjectAccessReview requests.
-	for _, at := range proxyAttrs {
-		authorized, reason, err = n.delegate.Authorize(ctx, at)
-		if err != nil {
-			return authorizer.DecisionDeny,
-				"AuthorizationError",
-				fmt.Errorf("authorization error (user=%s, verb=%s, resource=%s, subresource=%s): %w", at.GetName(), at.GetVerb(), at.GetResource(), at.GetSubresource(), err)
-		}
-		if authorized != authorizer.DecisionAllow {
-			return authorizer.DecisionDeny,
-				fmt.Sprintf("Forbidden (user=%s, verb=%s, resource=%s, subresource=%s): %s", at.GetName(), at.GetVerb(), at.GetResource(), at.GetSubresource(), reason),
-				nil
-		}
+	// This is the templated-resource attributes use-case.
+	var authorizers []authorizer.Authorizer
+	for _, proxyAttr := range proxyAttrs {
+		authorizers = append(authorizers, AuthFunc(func(ctx context.Context, _ authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+			return n.delegate.Authorize(ctx, proxyAttr)
+		}))
 	}
 
-	if authorized == authorizer.DecisionAllow {
-		return authorizer.DecisionAllow, "", nil
+	// Multiple attributes can occur when templating is configured in the
+	// configuration and a client specifies several templating values.
+	// Example:
+	// A client could specify 2 values for the templating in the query (if
+	// configured with TemplatedRessourceAttributes):
+	// 		bar and baz: ?foo=bar&foo=baz
+	//
+	// For the default use-case:
+	// | Value | Response |
+	// |-------|----------|
+	// | bar   | allow    |
+	// | baz   | deny     |
+	// the result of the union of the authorizers will deny the request.
+	// For the use-case with a network error:
+	// | Value | Response |
+	// |-------|----------|
+	// | bar   | allow    |
+	// | baz   | error    |
+	// the result of the union will be allow.
+	//
+	// This would not be an issue in a simple setup as the client defines the
+	// query params and could drop the baz value.
+	// BUT the templating variant is only useful if the upstream service
+	// interprets the query params as well.
+	// AND therefore upstream trusts in the verification of the query params.
```

Comment:
The issue is, that the use case is pretty edgy. Usually you bar those feature against huge `unsafe-feature-x` flags 😄
In sum it means that we can't accept the default behavior due to the rewrite use case.

Reply:
In sum it means that we can't accept the default behavior due to the rewrite use case and the comment highlight an example which emphasize this.

Reply:
I see what this is trying to say.

Union finishes on allow/deny. The original code checked _all_ are allowed, whereas with this change, it checks whether a single combination is allowed.

This change therefore looks like a bug?

Reply:
Oh, sorry. I think you lack the context:

We can't do that:
```go
return union.New(authorizers...).Authorize(ctx, nil)
```

We need to handle the error case like so:
```go
if err != nil {
		return authorizer.DecisionDeny, "error while authorizing", err
}
```.

The reasoning in the comments above is: why must not use just a union.

Reply:
That's different from what I'm trying to say.

You cannot use union at all. The union authorizer stops on the first allow. The previous code would only allow if and only if _all_ `proxyAttrs` would allow.